### DISCUSSION TITLE
Columns can't be a `NewType`, raises `TypeError: issubclass() arg 1 must be a class`

### DISCUSSION THREAD

**Discussion by vdozier-work** [fastapi/sqlmodel#981] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the SQLModel documentation, with the integrated search.
- [X] I already searched in Google "How to X in SQLModel" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to SQLModel but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to SQLModel but to [SQLAlchemy](https://github.com/sqlalchemy/sqlalchemy).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import string

from typing import NewType
from typing import Annotated
from collections.abc import Iterable
from pydantic import StringConstraints
from pydantic import AfterValidator
from sqlmodel import SQLModel
from sqlmodel import Field
from sqlmodel import create_engine

def _validate_charset(s: str, charset: Iterable[str]) -> str:
    illegal_characters = set(s).difference(set(charset))
    assert (
            illegal_characters == {}
    ), f'string "{s}" contains illegal characters: {illegal_characters}'
    return s

PrintableString = Annotated[str, StringConstraints(max_length=16), AfterValidator(lambda s: _validate_charset(s, string.printable))]

# Suppose I also want a type called SpecialString to ensure during type checking that
# plain old PrintableStrings don't get confused for a SpecialString
SpecialString = NewType("SpecialString", PrintableString)

class MyModel(SQLModel, table=True):
    id: int | None = Field(default=None, primary_key=True)
    special_data: SpecialString

sqlite_file_name = "database.db"
sqlite_url = f"sqlite:///{sqlite_file_name}"

if __name__ == "__main__":
    engine = create_engine(sqlite_url, echo=True)
    SQLModel.metadata.create_all(engine) # error! TypeError: issubclass() arg 1 must be a class
```

### Description

I am trying to set up a database with SQLModel where some fields use `typing.NewType`. This does not change the runtime type at all, it is just a way to make certain subtypes distinct to a typechecker like Mypy.

If I run the example code, `metadata.create_all()` produces an error on line 571 of `sqlmodel/main.py`. The issue seems to be that `issubclass()` does not recognize a `NewType` as a class. Currently, I can get around the problem with:

```python
# Suppose I also want a type called SpecialString to ensure during type checking that
# plain old PrintableStrings don't get confused for a SpecialString
if TYPE_CHECKING:
    SpecialString = NewType("SpecialString", PrintableString)
else:
    SpecialString = PrintableString
```

but that seems like explicitly checking `TYPE_CHECKING` kind of defeats the purpose of `typing.NewType`.

### Operating System

macOS

### SQLModel Version

0.0.19

### Python Version

3.12.3

**Comment by vdozier-work** [comment]
PS, looks like the same thing happens for `Literal`

**Comment by YuriiMotov** [comment]
You need to specify column type using `Field(sa_type=sa.String)`

**Reply by hellow554** [reply]
Well, while this works it defeats the purpose of defining the type in the first place, doesn't it?
I think a proper fix would be welcome.

But thanks for the workaround! :)
